sdk/core/Azure.Core.Experimental/tests/Variant/ValueUsageTests.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
One of the main design validation tests I would love to do is to develop a PropertyBag type; a custom map that maps from string (name) to Variant. It should be natural to use such bag, similarly to how DynamicData can be used, e.g. access a property in the bag and compare it's value to some literal.

```csharp
PropertyBag bag = ...
float x = bag["foo"] == 5 ? bag["bar"] : 0;
dynamic d = bag;
float y = d.foo == 5 ? d.bar : 0;
bag.Serialize(...);
// etc.
```

Reply:
Thanks!  I'll include a prototype for this in the tests.  Although, do you want it to be a type in Azure.Core.Experimental for now, as something that would release with Value/Variant?

Reply:
@KrzysztofCwalina, the following succeeds in this PR:
https://github.com/Azure/azure-sdk-for-net/pull/38103/files#diff-c58c3910bb3006786a982dda10610efca07f7ac58129129ee222bd9fd0546d66R142-R151

It uses `int` instead of `float`.  Are you concerned with being able to cast across numeric values, or is it ok for you if we throw invalid cast if the type expected by the call site isn't the same as the type inside the Value?